File: portal_estagios_unialfa-main/portal-php/classes/Candidatura.php
<?php

class Candidatura
{
    private $id;
    private $vagaId;
    private $alunoId;
    private $status;

    public function __construct($id, $vagaId, $alunoId, $status)
    {
        $this->id = $id;
        $this->vagaId = $vagaId;
        $this->alunoId = $alunoId;
        $this->status = $status;
    }

    public function getId()
    {
        return $this->id;
    }

    public function getVagaId()
    {
        return $this->vagaId;
    }

    public function getAlunoId()
    {
        return $this->alunoId;
    }

    public function getStatus()
    {
        return $this->status;
    }
}

File: src/main/java/com/unialfa/model/Candidatura.java
package com.unialfa.model;

public class Candidatura {

    private int id;
    private String status;

    private Integer alunoId;
    private Integer vagaId;

    // 🔥 CAMPOS PARA EXIBIÇÃO (JOIN)
    private String alunoNome;
    private String vagaTitulo;

    public Candidatura() {
    }

    public Candidatura(int id, String status, Integer alunoId, Integer vagaId, String alunoNome, String vagaTitulo) {
        this.id = id;
        this.status = status;
        this.alunoId = alunoId;
        this.vagaId = vagaId;
        this.alunoNome = alunoNome;
        this.vagaTitulo = vagaTitulo;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Integer getAlunoId() {
        return alunoId;
    }

    public void setAlunoId(Integer alunoId) {
        this.alunoId = alunoId;
    }

    public Integer getVagaId() {
        return vagaId;
    }

    public void setVagaId(Integer vagaId) {
        this.vagaId = vagaId;
    }

    public String getAlunoNome() {
        return alunoNome;
    }

    public void setAlunoNome(String alunoNome) {
        this.alunoNome = alunoNome;
    }

    public String getVagaTitulo() {
        return vagaTitulo;
    }

    public void setVagaTitulo(String vagaTitulo) {
        this.vagaTitulo = vagaTitulo;
    }
}